### PR TITLE
Utilities: Various changes in passwd

### PR DESCRIPTION
**Pull request by brapru**
This adds additional functionalities to the passwd utility:

-  Requires non-root users to authenticate with the current password before being able to change to new password
-  Prompts the user to enter the new password twice to check for mismatches
-  Returns error when attempting to change password with no input
-  Provides more verbose output indicating the status of the changed password

### CODE REVIEW

File: Userland/Utilities/passwd.cpp [hunk 1]
```diff
@@ -84,6 +84,14 @@ int main(int argc, char** argv)
     } else if (unlock) {
         target_account.set_password_enabled(true);
     } else {
+        if (current_uid != 0) {
+            auto old_password = Core::get_password("Old password: ");
+            if (!target_account.authenticate(old_password.value().characters())) {
```
Comment (awesomekling) [inline Userland/Utilities/passwd.cpp]: Missing an error check here (`old_password.is_error()`)

File: Userland/Utilities/passwd.cpp [hunk 2]
```diff
@@ -85,14 +85,20 @@ int main(int argc, char** argv)
         target_account.set_password_enabled(true);
     } else {
         if (current_uid != 0) {
-            auto old_password = Core::get_password("Old password: ");
-            if (!target_account.authenticate(old_password.value().characters())) {
+            auto current_password = Core::get_password("Current password: ");
+            if (!target_account.authenticate(current_password.value().characters())) {
```
Comment (awesomekling) [inline Userland/Utilities/passwd.cpp:89]: Missing error check.

File: Userland/Utilities/passwd.cpp [hunk 3]
```diff
@@ -85,14 +85,20 @@ int main(int argc, char** argv)
...
                 warnln("Incorrect or disabled password.");
                 return 1;
             }
         }
 
         auto new_password = Core::get_password("New password: ");
+        auto new_password_retype = Core::get_password("Retype new password: ");
+        if (new_password.value() != new_password_retype.value()) {
```
Comment (awesomekling) [inline Userland/Utilities/passwd.cpp]: Missing error check.

File: Userland/Utilities/passwd.cpp [hunk 4]
```diff
@@ -95,6 +95,13 @@ int main(int argc, char** argv)
 
         auto new_password = Core::get_password("New password: ");
         auto new_password_retype = Core::get_password("Retype new password: ");
+
+        if (new_password.value().is_empty() && new_password_retype.value().is_empty()) {
```
Comment (awesomekling) [inline Userland/Utilities/passwd.cpp:99]: Neither of the above values have been error checked.

**Review by awesomekling — CHANGES_REQUESTED** [review]

File: Userland/Utilities/passwd.cpp (hunk 4 above)
Comment (brapru) [reply Userland/Utilities/passwd.cpp:113]: Changes made.

File: Userland/Utilities/passwd.cpp [hunk 5]
```diff
@@ -111,10 +112,12 @@ int main(int argc, char** argv)
 
         if (new_password.value() != new_password_retype.value()) {
             warnln("Sorry, passwords don't match.");
+            warnln("Password for user {} unchanged.", target_account.username());
             return 1;
         }
 
         target_account.set_password(new_password.value().characters());
+        outln("Password for user {} successfully updated.", target_account.username());
```
Comment (awesomekling) [inline Userland/Utilities/passwd.cpp:126]: This isn't true yet. It wasn't updated until after the call to `sync()` below succeeds.

Comment (awesomekling) [reply Userland/Utilities/passwd.cpp:126]: Looks great other than this one thing :)

Comment (brapru) [reply Userland/Utilities/passwd.cpp]: Made the change 👍

**Pull request merged**